runner.go

```diff
@@ -312,16 +311,15 @@ func process(mgr *Manager, idx int) {
 				_ = mgr.with(func(c *faktory.Client) error {
 					return c.Fail(job.Jid, fmt.Errorf("No handler for %s", job.Type), nil)
 				})
+				mgr.activityMutex.Unlock()
```

Comment:
I'm not sure this approach would work because this means we can only process one job at a time per velen worker!

A potential solution could be to move away from the mutex, rework how we compute the activity, and maybe even use https://golang.org/pkg/sync/#Map.

If you want to discuss this, let me know!

Reply:
Following our discussion on Basecamp, are we good to merge?